p2p/trust/trustmetric_test.go

```diff
@@ -210,7 +210,10 @@ func TestTrustMetricStopPause(t *testing.T) {
 	// Give the pause some time to take place
 	time.Sleep(10 * time.Millisecond)
 
+	tm.mtx.Lock()
```

Comment:
Perhaps we can introduce a copy function that locks to create a copy of the `tm` and returns that. then we can just do eg. `first := tm.Copy().numIntervals`